Fix detection of changed files in CI

zanieb opened the pull request:
These were literals instead of expressions... and were consequently not evaluated.

Fixes bug from #8225

MichaReiser commented:
I'm not sure if that's the root cause because the checks did ran in the last few days and the [`if`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif) documentation mentions that omitting the `${{ }}` is okay except when the expression starts with `!`.

zanieb commented:
@MichaReiser they didn't run on this branch before the change, but did after 🤷‍♀️ perhaps a change on their end

zanieb commented:
Also, I ran the workflow with debug logs and the "Determine changes" workflow is not at fault

zanieb commented:
Hm but it _does_ seem to be at fault over in the other branch https://github.com/astral-sh/ruff/actions/runs/7122120523/job/19395730196

zanieb commented:
```
 M	Cargo.toml
  ##[debug]All diff files: {"A":[],"C":[],"D":[],"M":["Cargo.toml"],"R":[],"T":[],"U":[],"X":[]}
  All Done!
  ::endgroup::
::group::changed-files-yaml-linter
changed-files-yaml-linter
::group::changed-files-yaml-formatter
changed-files-yaml-formatter
::group::changed-files-yaml-code
changed-files-yaml-code
  ##[debug]All filtered diff files for code: {"A":[],"C":[],"D":[],"M":[],"R":[],"T":[],"U":[],"X":[]}
  ##[debug]Dir names include file patterns: []
  ##[debug]Added files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]Copied files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]Modified files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]Renamed files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]Type changed files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]Unmerged files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]Unknown files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]All changed and modified files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]All changed files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]All other changed files: {"paths":"Cargo.toml","count":"1"}
  ##[debug]other_changed_files: ["Cargo.toml"]
  ##[debug]Dir names include file patterns: []
  ##[debug]All modified files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]other_modified_files: ["Cargo.toml"]
  ##[debug]Dir names include file patterns: []
  ##[debug]Deleted files: {"paths":"","count":"0"}
  ##[debug]Dir names include file patterns: []
  ##[debug]other_deleted_files: []
```

ofek commented:
Now that the root cause has been fixed you may want to remove the `${{ }}` again to look idiomatic

zanieb commented:
Eh we use `${{ }}` for every other `if` statement in that file. I'd prefer to be consistent.
